spinn_machine/virtual_machine.py

```diff
@@ -109,7 +108,7 @@ def _verify_width_height(width, height):
 
 def virtual_machine(
         width=None, height=None, with_wrap_arounds=None, version=None,
-        n_cpus_per_chip=Machine.MAX_CORES_PER_CHIP, with_monitors=True,
+        n_cpus_per_chip=None,
```

Comment:
realign params to not have such long empty spaces

Reply:
on a further thing (i know its not speicfic to the pr. but could we add or None to the docs for with-wrap_arounds and version please. its flagging elsewhere that its getting None put in. thanks

Reply:
params realigned. Agreed.

Reply:
I am actually highly highly tempted to completely NUKE with_wrap_arounds and version params and to just use the width and height values.

Reply:
i dont think you can nuke them from a virtual machine. we need to still support spinn3 virtual, and you need to be able to support a toriod and none toriod virtual machine. 

renemebr that vm is also used by end suers who have a machine they plan to operate on, but hasnt got it at hand and are debugging their scripts trhough mapping etc

Reply:
If you give height 2 and width 2 it knows its a spin3

If you give width 12 and height 12 it knows it is a three board toriod
Where width 16 and height 16 reflect a three board none toroid.

All version ever does is fill in width and height or go boom if you did wrong.

All with_wrap_arounds ever does is go boom if you got it wrong.  Plus a boolean value can only distinguish between two of the 4 possible cases.

Reply:
are you sure that every width and height combination can identify if it has wrap arounds or not?

i would try to argue 2 by 2 is not only a spinn3 but a weird board....butt hen weird board....bugger..... if you can prove all you need is height and width to know wraps or not. then fine.

Reply:
Excluding 2 x 2

The current rule is if wraparound specified bevel the request else if an axis is divisible by 12 it is a wrap else it is not a wrap.

In practice the rule could be
If the axis is not a multiple of 4 it is fictional so no wrap.
If the axis is a multiple of 12 wrap
If the axis is 8 no wrap
If the axis is 4 less is a multiple of 12 no wrap

That leaves the axis is 4 more than a multiple of 12.
To the best if my knowledge that would have to be non wrap as without the toriod shape the wrapped chips don't fit on the other side.

So the question is do we want to support incorrect wraparounds?

Reply:
to which the answer should be no. i think. so ok. prune ahead